### ISSUE TITLE
Don't use set union in appending nodes to Term?

### ISSUE THREAD

**Issue by dchandekstark** [ActiveTriples/ActiveTriples#42]
The use case is I want to append nodes to a term only if they are not isomorphic (RDF::Isomorphic) with an existing node.  The `<<` append method in Term uses set union on two Arrays, which relies on `eql?` and `hash` for comparison.  It would be convenient if the method filtered out values using `include?`, which uses `==` for comparison, and so is more safely overridden in a Resource class (I think).

Another possibility I suppose is to fork how appending behaves based on whether the Resource class implements RDF::Isomorphic and/or an explicit option is given (`isomorphic: true` or something) in the property definition.

I am able to work around this (I would not call it a bug), but I'm asking the question to see what folks think.

**Comment by no-reply** [comment]
I'm looking now at the viability of using the new `rdf-normalize`to better (and cheaply) handle isomorphism on `set` and `delete`. I'll report back with some examples of how things work under that regime, and hopefully some benchmarks so we can decide whether that approach will work as a default handling.

**Comment by no-reply** [comment]
I'm thinking this should be scoped to a refactor of how `#<<`, `#set`, and (private) `#set_value` interact.  After #226, I don't think there's any reason these operations ever need to deal with `Array` representations. It should be possible to align this all much more closely with RDF.rb's `RDF::Term` equality.

My concern with isomorphism (or normalization) is that we would need to define the scope of the graph to check for isomorphism, and it would likely change across persistence strategies.

For example, given: 

```turtle
# in `s1`'s graph
<s1> <p> _:o .

# in `_:o`'s graph
_:o dc:title "o" .
```

...setting with `s1.p1 << RDF::Node.new` would be a no-op under isomorphism on `s1`'s graph.

Add to this the potential for EXP complexity in isomorphism checks, and I think we should just punt this.

**Comment by no-reply** [comment]
Calling this closed by #226, as discussed above.
